### PR TITLE
feat: withdraw claim (CMC-71)

### PR DESCRIPTION
**Pull request by swalker125**
### JIRA link (if applicable) ###

https://tools.hmcts.net/jira/browse/CMC-71

### Change description ###

Withdraw claim event to move case to closed state from created state

**Does this PR introduce a breaking change?** (check one with "x")

```
[ ] Yes
[x] No
```

### CODE REVIEW

File: ccd-definition/CaseEvent.json [hunk 1]
```diff
@@ -123,5 +123,19 @@
...
     "EndButtonLabel": "Submit your response"
+  },
+  {
+    "LiveFrom": "01/01/2017",
+    "CaseTypeID": "UNSPECIFIED_CLAIMS",
+    "ID": "WITHDRAW_CLAIM",
+    "Name": "Withdraw claim",
+    "Description": "Withdraw a claim",
+    "DisplayOrder": 9,
+    "PreConditionState(s)": "CREATED",
```
Comment (dharmendrak) [inline ccd-definition/CaseEvent.json]: pre condition can be any state

File: ccd-definition/CaseEventToComplexTypes/WithdrawClaim.json [hunk 2]
```diff
@@ -0,0 +1,24 @@
...
+    "CaseEventID": "WITHDRAW_CLAIM",
+    "CaseFieldID": "withdrawClaim",
+    "ListElementCode": "date",
+    "EventElementLabel": "Enter the date from which you want the claim to be withdrawn",
+    "EventHintText": "The date must be today's date or earlier",
+    "FieldDisplayOrder": 1,
+    "DisplayContext": "MANDATORY"
+  },
+  {
+    "LiveFrom": "01/01/2017",
```
Comment (dharmendrak) [inline ccd-definition/CaseEventToComplexTypes/WithdrawClaim.json:14]: reason is suppose to be optional

File: ccd-definition/State.json [hunk 3]
```diff
@@ -22,5 +22,13 @@
     "Name": "Awaiting claimant intention",
     "Description": "Awaiting claimant intention",
     "DisplayOrder": 3
+  },
+  {
+    "LiveFrom": "01/01/2017",
+    "CaseTypeID": "UNSPECIFIED_CLAIMS",
+    "ID": "CLOSED",
+    "Name": "Closed",
+    "Description": "Case is withdrawn",
```
Comment (dharmendrak) [inline ccd-definition/State.json]: should it be `case is closed`

File: ccd-definition/CaseEvent.json (hunk 1 above)
Comment (swalker125) [reply ccd-definition/CaseEvent.json]: ticket doesn't mention any state,  I have taken this Scenario - draft claim has been issued and case is not closed as CREATED state

File: ccd-definition/CaseEventToComplexTypes/WithdrawClaim.json (hunk 2 above)
Comment (swalker125) [reply ccd-definition/CaseEventToComplexTypes/WithdrawClaim.json:14]: you're right

File: ccd-definition/State.json (hunk 3 above)
Comment (swalker125) [reply ccd-definition/State.json]: no content for what this should be

File: ccd-definition/CaseEvent.json (hunk 1 above)
Comment (dharmendrak) [reply ccd-definition/CaseEvent.json]: based on prototype as approved by Darren, pre condition can be at any state

Comment (swalker125) [reply ccd-definition/CaseEvent.json]: updated

**Review by dharmendrak — DISMISSED** [review]

**Review by pd-dudkiewicz — DISMISSED** [review]

**Review by douglasrice — APPROVED** [review]

**Review by pd-dudkiewicz — APPROVED** [review]

**Pull request merged**